### PR TITLE
Bug 1690496 - Make init works _exactly_ like in glean-core

### PR DESCRIPTION
**Pull request by brizental**

### CODE REVIEW

File: src/glean.ts [hunk 1]
```diff
@@ -189,17 +200,43 @@ class Glean {
...
+      if (clientId) {
+        if (clientId !== KNOWN_CLIENT_ID) {
+          // Temporarily enable uploading so we can submit a
+          // deletion request ping.
+          Glean.uploadEnabled = true;
+          await Glean.onUploadDisabled();
+        }
+      } else {
+        await Glean.clearMetrics();
+      }
```
Comment (Dexterp37) [inline src/glean.ts:239]: Should this logic be refactored out?

**Review by Dexterp37 — APPROVED** [review]
r+wc

File: src/glean.ts (hunk 1 above)
Comment (brizental) [reply src/glean.ts:239]: Honestly, I'd rather leave it here for now. If init starts getting too big I can break it off. Would that be alright?

Comment (Dexterp37) [reply src/glean.ts:239]: Works for me

**Pull request merged**